### ISSUE TITLE
A convenient way to renew domains.

### ISSUE THREAD

**Issue by yetanotherusernamebecausegithubisbugged** [namecoin/namecoin#223]
Manually renew a domain every 6 months is just inconvenient. If you want somebody to confirm he still alive and needs that domain, it should be as easy as sending NMC to some address, not opening the program and clicking buttons. And the term must be at least a year because six months is just too fast.

**Comment by JeremyRand** [comment]
This is multiple issues.  I've created an issue for the renewal period: https://github.com/namecoin/meta/issues/16 (we were already talking about this, for reason it didn't wind up as a GitHub ticket).

As for your other issue, I don't understand.  How is "sending NMC to some address" easier than "opening the program and clicking buttons or messing with the daemon"?  The latter is required to do the former.  There is some ongoing work to allow you to delegate renewal permissions to a third party if that's what you want.  There's also ongoing work to allow automatically renewing domains if the client stays open (or is opened periodically).  Other than that, I have no idea what exactly you're proposing.  Can you elaborate?

**Comment by yetanotherusernamebecausegithubisbugged** [comment]
> Can you elaborate?

People are lazy and in order for them to adopt new technology it must be as easy as possible. The way I renew my clearnet domains: I send money to a registrar every year. Not clicking any buttons, not even going to a registrar website. The same is doing many people I know a lot of which are not tech-savvy. And even with that simplicity they're still running into a 'forget to pay' problem. It's obvious that people will lose their wallet.dat's also and there is not so much that can be done about that. 

But it is possible for namecoin at least to keep the same simplicity and even to make the process more secrue. Use case I'm talking about is:

To renew a .bit domain user is going (or asking somebody to go) to an exchange site, buys NMC and withdraws it to a special address domains binded to. The network sees that there was some activity on that addres, user is alive and still wants his domains, and renew the binded domains automatically.

That is as simple and reliable as possible. Also if the user has lose his wallet.dat (of course he will know it when there is a time comes to renew a domain, not earlier), he still can renew his domains to have some time to register a new domains, warn sites visitors, set a redirections, etc. This will make things both more secure and convenient. 

Also (I don't know though maybe it's already implemented) it would be good to have some expiration period, a month or 1.5 month, because a lot of people are realizing that their domain is not paid only when their site stops working. If they'll lose domain that way they will blame the namecoin not themselves.

**Comment by JeremyRand** [comment]
I think I understand your use case -- a user doesn't want his/her private keys to be required to renew a name.  The renewal delegation feature I mentioned previously would allow the renewal permission to be set to "anyone can spend".  This would basically mean that anyone who possesses some NMC could renew that name, including a web service.  I think this partially addresses your use case.  Actually making the renewal contingent on coins being sent to an arbitrary address would be quite a bit harder, I'm not sure if the script system can even handle such things.

I think we will add a feature which makes names stop resolving some time before the name expires -- this will give name owners a warning before they lose control of their name.  It's not a namecoind feature; this fits better as an NMControl feature.  (End users will see the same result -- domain names that are close to expiring will result in an error indicating that the name is about to expire, instead of the website loading.)

Does this address your concerns acceptably?

**Comment by JeremyRand** [comment]
FYI, the proposal for making names stop resolving before they are expired is at https://wiki.namecoin.info/index.php?title=User:Biolizard89/ExpireWarning

**Comment by yetanotherusernamebecausegithubisbugged** [comment]
> Does this address your concerns acceptably?

Yes that definitely covers everything I'm concerned about right now. And thank you for making the internet a better place.

**Comment by JeremyRand** [comment]
Sounds good -- and thank you for raising these concerns.
